polaris/curation/viz_utils.py

```diff
@@ -102,3 +140,200 @@ def check_undefined_stereocenters(dataset: PandasDataFrame):
     )
     fig = dm.to_image(to_plot.mol.tolist(), legends=legends)
     return fig
+
+
+def visualize_chemspace(
+    data: pd.DataFrame,
+    mol_col: str = "smiles",
+    size_col=None,
+    seed=1428,
+    umap_metric="jaccard",
+    hue_names: List[str] = None,
+):
+    """
+    Visualize the chemical space by doing a dimensionality reduction with UMAP.
+
+    Args:
+        data: a dataframe
+        split_names: names of the types of splits, should be found in columns of the "data" dataframe
+        mol_col: string column where we will look for a molecule to featurize
+        size_col: used for setting the style of the resulting scatterplot
+        seed: int, random seed to use for reproducibility
+        umap_metric: similarity metric to use for constructing the UMAP representation
+        hue_names: List of grouping variable that will produce points with different colors for the same UMAP plotting.
+
+    Returns:
+        figs, figures for each kind of split specified in split_names
+    """
+    figs = plt.figure(num=3)
```

Comment:
Why `num=3`? I think we should do something like:

```python
hue_cols = [None] if hue_cols is None else hue_cols

num_figs = len(hue_cols)
fig, axs = plt.subplots(nrows=1, ncols=num_figs, figsize=(fig_size * num_figs, fig_size))
axs = np.atleast_1d(axs)

for ax, hue_col in zip(axs, hue_cols): 
    sns.plot(..., ax=ax, hue=hue_col)
```

Where `fig_size` is also a parameter.